package/lib/src/modal/modal_sheet.dart

```diff
@@ -155,6 +167,8 @@ mixin ModalSheetRouteMixin<T> on ModalRoute<T> {
 
   Widget buildContent(BuildContext context);
 
+  Widget? get barrierWidget;
```

Comment:
I think this should be a builder rather than a widget because the barrier is lazily built in `buildModalBarrier`. The signature could be something like `ModalSheetBarrierBuilder? get barrierBuilder` where `ModalSheetBarrierBuilder` is:
```dart
typedef ModalSheetBarrierBuilder = Widget Function(PageRoute route, VoidCallback onDismissCallback);
```

The `route` parameter is necessary to get `barrierColor`, `barrierDismissible`, etc...